health/healthd_common.cpp

```diff
@@ -211,15 +222,152 @@ static void wakealarm_init(void) {
+        s_props.batteryStatus = BATTERY_STATUS_UNKNOWN;
+    else if(!strncmp((char *)status, "Charging", 8))
+        s_props.batteryStatus = BATTERY_STATUS_CHARGING;
+    else if(!strncmp((char *)status, "Discharging", 11))
+        s_props.batteryStatus = BATTERY_STATUS_DISCHARGING;
+    else if(!strncmp((char *)status, "Not charging", 12))
+        s_props.batteryStatus = BATTERY_STATUS_NOT_CHARGING;
+    else if(!strncmp((char *)status, "Full", 4))
+        s_props.batteryStatus = BATTERY_STATUS_FULL;
+}
+
+static void parse_battery_health(uint8_t *health)
+{
+    if(!strncmp((char *)health, "Unknown", 7))
+        s_props.batteryHealth = BATTERY_HEALTH_UNKNOWN;
+    else if(!strncmp((char *)health, "Good", 4))
+        s_props.batteryHealth = BATTERY_HEALTH_GOOD;
+    else if(!strncmp((char *)health, "Overheat", 8))
+        s_props.batteryHealth = BATTERY_HEALTH_OVERHEAT;
+    else if(!strncmp((char *)health, "Dead", 4))
+        s_props.batteryHealth = BATTERY_HEALTH_DEAD;
+    else if(!strncmp((char *)health, "Over voltage", 12))
+        s_props.batteryHealth = BATTERY_HEALTH_OVER_VOLTAGE;
+    else if(!strncmp((char *)health, "Unspecified failure", 18))
+        s_props.batteryHealth = BATTERY_HEALTH_UNSPECIFIED_FAILURE;
+    else if(!strncmp((char *)health, "Cold", 4))
+        s_props.batteryHealth = BATTERY_HEALTH_COLD;
+}
+
+static void parse_battery_type(uint8_t *type)
+{
+    if (!strncmp((char *)type, "Mains", 7))
+        s_props.chargerAcOnline = true;
+    else if (!strncmp((char *)type, "USB", 3))
+        s_props.chargerUsbOnline = true;
+}
+
+static void parse_init_properties(struct initial_pkt *ipkt)
+{
+    s_props.batteryTechnology = (char *)ipkt->technology;
+    s_props.batteryPresent = (strncmp((char *)ipkt->present, "1", 1) == 0);
+    parse_battery_type(ipkt->type);
+}
+
+static void parse_battery_properties(struct monitor_pkt *mpkt)
+{
+    s_props.batteryLevel = mpkt->capacity;
+    s_props.batteryVoltage = mpkt->voltage_now;
+    s_props.batteryTemperature = mpkt->temp;
+    s_props.batteryFullCharge = mpkt->charge_full;
+    s_props.batteryChargeCounter = mpkt->charge_now;
+    parse_battery_status(mpkt->status);
+    parse_battery_health(mpkt->health);
+}
+
+static int connect_vsock(int *vsock_fd) {
+    struct sockaddr_vm sa = {
+        .svm_family = AF_VSOCK,
+        .svm_cid = VMADDR_CID_HOST,
+        .svm_port = HEALTH_PORT,
+    };
+    *vsock_fd = socket(AF_VSOCK, SOCK_STREAM, 0);
+    if (*vsock_fd < 0) {
+        KLOG_WARNING(LOG_TAG, "healthd socket init failed\n");
+        return *vsock_fd;
+    }
+    if (connect(*vsock_fd, (struct sockaddr*)&sa, sizeof(sa)) != 0) {
+            KLOG_WARNING(LOG_TAG, "healthd connect failed\n");
+            close(*vsock_fd);
+	    return -1;
+    }
+    KLOG_INFO(LOG_TAG, "healthd connect to cid(%d) port(%d)\n", sa.svm_cid, sa.svm_port);
+    return 0;
+}
+
+static void recv_vsock(int *vsock_fd) {
+    char msgbuf[1024];
```

Comment:
shall initialize.

Reply:
Am doing memset everytime before passing it to recv command. So, should be fine I guess.